src/gtest/test_coins.cpp

```diff
@@ -831,3 +977,492 @@ TEST(CoinsTests, AnchorsTest)
+                }
+                break;
+            }
+        }
+
+        // Sanity checks for all cases
+        cache2.Flush();
+        cache1.Flush();
+        auto latest = base.GetLatestSubtree(type);
+        if (latest.has_value()) {
+            auto latest2 = cache1.GetLatestSubtree(type);
+            ASSERT_TRUE(latest2.has_value());
+            EXPECT_EQ(latest->index, latest2->index);
+            EXPECT_EQ(latest->root, latest2->root);
+            EXPECT_EQ(latest->nHeight, latest2->nHeight);
+        }
+
+        delete base.memorydb;
+    };
+
+    attempt({Push0, Push1, FlushCache2, Get0C2});
+    attempt({Push0, FlushCache2, FlushCache1, FlushCache1, LatestBaseEquals0});
+    attempt({None0B, None0C1, None0C2, LatestBaseEqualsX, LatestC1EqualsX, LatestC2EqualsX});
+    attempt({Push0, None0B, None0C1, Get0C2, None1C2});
+    attempt({Push0, None0C1, FlushCache2, Get0C2, Get0C1, LatestC1Equals0, LatestC2Equals0, LatestBaseEqualsX});
+    attempt({Push0, None0C1, FlushCache2, Get0C2, Get0C1, None0B, FlushCache1, Get0B, LatestC1Equals0, LatestC2Equals0, LatestBaseEquals0});
+    attempt({Push0, Push1, Get0C2, None0C1, FlushCache2, Get0C2, Get0C1, Get1C2, Get1C1, None0B, FlushCache1, Get0B, Get1B});
+    attempt({Push0, FlushCache2, Pop, None0C2, Get0C1});
+    attempt({Push0, Push1, FlushCache2, Push2, Pop, Get0C2, Get1C2, LatestC2Equals1, Pop, Get0C2, LatestC2Equals0, Pop, FlushCache2, None0C1, None0C2, LatestC1EqualsX});
+    attempt({Push0, FlushCache2, Push1, FlushCache2, FlushCache1, Push2, Push3, Pop, FlushCache2, LatestC2Equals2, LatestC1Equals2, LatestBaseEquals1, FlushCache1, LatestBaseEquals2});
+    attempt({Push0, Push1, FlushCache2, Get0C1, Pop, Pop, FlushCache2, None0C1, None0C2, LatestC1EqualsX, LatestC2EqualsX});
+    attempt({Push0, Push1, FlushCache2, LatestC1Equals1, Pop, FlushCache2, LatestC1Equals0});
+    attempt({Push0, FlushCache2, Push1, None2C2, None2C1});
+    attempt({Push0, FlushCache2, Push1, Get1C2});
+    attempt({Push0, Push1, Get0C2, Get1C2, FlushCache2, Get0C2, Get0C2, Get1C2, Get1C2});
+    attempt({Push0, Push1, LatestC2Equals1, FlushCache2, Push2, LatestC2Equals2});
+    attempt({Push0, FlushCache2, LatestC2Equals0, LatestC1Equals0});attempt({None0B, None0C1, None0C2, LatestBaseEqualsX, LatestC1EqualsX, LatestC2EqualsX});
+    attempt({Push0, None0B, None0C1, Get0C2, None1C2});
+    attempt({Push0, None0C1, FlushCache2, Get0C2, Get0C1, LatestC1Equals0, LatestC2Equals0, LatestBaseEqualsX});
+    attempt({Push0, None0C1, FlushCache2, Get0C2, Get0C1, None0B, FlushCache1, Get0B, LatestC1Equals0, LatestC2Equals0, LatestBaseEquals0});
+    attempt({Push0, Push1, Get0C2, None0C1, FlushCache2, Get0C2, Get0C1, Get1C2, Get1C1, None0B, FlushCache1, Get0B, Get1B});
+    attempt({Push0, FlushCache2, Pop, None0C2, Get0C1});
+    attempt({Push0, Push1, FlushCache2, Push2, Pop, Get0C2, Get1C2, LatestC2Equals1, Pop, Get0C2, LatestC2Equals0, Pop, FlushCache2, None0C1, None0C2, LatestC1EqualsX});
```

Comment:
```suggestion
    attempt({Push0, FlushCache2, LatestC2Equals0, LatestC1Equals0});
```
The deleted lines are duplicates of lines 1420 to 1426 inclusive.